File: src/main/java/um/haberes/core/hexagonal/cursos/cargo_tipo/domain/model/CargoTipo.java
package um.haberes.core.hexagonal.cursos.cargo_tipo.domain.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CargoTipo {

    private Integer cargoTipoId;

    @Builder.Default
    private Byte aCargo = 0;

    @Builder.Default
    private String nombre = "";

    @Builder.Default
    private int precedencia = 0;
}

File: src/main/java/um/haberes/core/kotlin/model/CargoTipo.kt
package um.haberes.core.kotlin.model

import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table
data class CargoTipo(

    @Id
    var cargoTipoId: Int? = null,

    var aCargo: Byte = 0,
    var nombre: String = "",
    var precedencia: Int = 0

) : Auditable()
